python/pyarrow/table.pxi

```diff
@@ -1177,13 +1177,13 @@ cdef class RecordBatch(_PandasConvertible):
         pyarrow.RecordBatch
         """
         from pyarrow.pandas_compat import dataframe_to_arrays
-        arrays, schema = dataframe_to_arrays(
+        arrays, schema, n_rows = dataframe_to_arrays(
             df, schema, preserve_index, nthreads=nthreads, columns=columns
         )
-        return cls.from_arrays(arrays, schema=schema)
+        return cls.from_arrays(arrays, schema=schema, n_rows=n_rows)
 
     @staticmethod
-    def from_arrays(list arrays, names=None, schema=None, metadata=None):
+    def from_arrays(list arrays, names=None, schema=None, metadata=None, n_rows=None):
```

Comment:
If we keep the keyword, I would also make this `num_rows` to keep it consistent with the public property of a Table. 
But, another option is also to handle this `n_rows==0` case in `from_pandas`. Now, I suppose in theory it can be useful to expose here, to make it possible to create an empty table with a given number of rows via the Python API (although is is very much an esoteric use case)

Reply:
To sum up the conclusion from our talk today: there would be added confusion and not much value in keeping `n_row` exposed in `from_arrays`. I will move the code to `from_pandas`.

I am figuring out the Cython code and will commit as soon as I am finished.